RadixWallet/Clients/OnLedgerEntitiesClient/OnLedgerEntitiesClient+Interface.swift

```diff
@@ -112,46 +112,104 @@ extension OnLedgerEntitiesClient {
 
 extension OnLedgerEntitiesClient {
 	@Sendable
-	public func getEntity(
-		_ address: Address,
+	public func getEntities(
+		addresses: [Address],
 		metadataKeys: Set<EntityMetadataKey>,
-		forceRefresh: Bool = false
-	) async throws -> OnLedgerEntity {
-		guard let resource = try await getEntities([address], metadataKeys, nil, forceRefresh).first else {
-			throw Error.emptyResponse
-		}
-		return resource
+		forceRefresh: Bool = false,
+		atLedgerState: AtLedgerState? = nil
+	) async throws -> [OnLedgerEntity] {
+		try await getEntities(
+			addresses,
+			metadataKeys,
+			atLedgerState,
+			forceRefresh
+		)
 	}
 
 	@Sendable
 	public func getAccounts(
 		_ addresses: [AccountAddress],
-		forceRefresh: Bool = false
+		metadataKeys: Set<EntityMetadataKey> = .resourceMetadataKeys,
+		forceRefresh: Bool = false,
+		atLedgerState: AtLedgerState? = nil
 	) async throws -> [OnLedgerEntity.Account] {
-		try await getEntities(addresses.map(\.asGeneral), .resourceMetadataKeys, nil, forceRefresh).compactMap(\.account)
+		try await getEntities(
+			addresses: addresses.map(\.asGeneral),
+			metadataKeys: metadataKeys,
+			forceRefresh: forceRefresh,
+			atLedgerState: atLedgerState
+		).compactMap(\.account)
+	}
+
+	@Sendable
+	public func getEntity(
+		_ address: Address,
+		metadataKeys: Set<EntityMetadataKey>,
+		forceRefresh: Bool = false,
+		atLedgerState: AtLedgerState? = nil
+	) async throws -> OnLedgerEntity {
+		guard
+			let entity = try await getEntities(
+				addresses: [address],
+				metadataKeys: metadataKeys,
+				forceRefresh: forceRefresh,
+				atLedgerState: atLedgerState
+			)
+			.first
+		else {
+			throw Error.emptyResponse
+		}
+
+		return entity
 	}
 
 	@Sendable
 	public func getAccount(
 		_ address: AccountAddress,
-		metadataKeys: Set<EntityMetadataKey> = .resourceMetadataKeys
+		metadataKeys: Set<EntityMetadataKey> = .resourceMetadataKeys,
+		forceRefresh: Bool = false,
+		atLedgerState: AtLedgerState? = nil
 	) async throws -> OnLedgerEntity.Account {
-		guard let account = try await getEntity(address.asGeneral, metadataKeys: metadataKeys).account else {
+		guard
+			let account = try await getEntity(
```

Comment:
Why the line break, here and above?

Reply:
I suppose this could be simplified a bit, by having the return statement inside a do catch? `return try await getEntity...`